app/controllers/CyaMaintainProfileController.scala

```diff
@@ -104,6 +103,38 @@ class CyaMaintainProfileController @Inject() (
       case Left(errors) =>
         Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
     }
+  }
+
+  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    TraderProfile.validateNirmsNumber(request.userAnswers) match {
```

Comment:
you can use buildNirms here instead of validate nirms

Reply:
I will check if I can use buildNirms instead :)